src/compiler/diagnosticMessages.json

```diff
@@ -3699,6 +3699,10 @@
         "category": "Error",
         "code": 2861
     },
+    "Type '{0}' is generic and can only be indexed for reading.": {

```

Comment:
I know we hate the existing message about "T could be instantiated with a different subtype", buuuuuuuuuuuut..what about:

Type {property access} cannot be used to index '{type variable}' because '{type variable}' could be instantiated with an index signature that is a subtype of '{index signature}'.

Reply:
I don't know if/how adding that explanation helps people... Also, mentioning the index type makes it so you get two almost identical error messages if the index type is a union (e.g. `string | symbol`), which is why Anders suggested above to omit the index type entirely.
Why do we hate the existing "T could be instantiated with a different subtype"?

Reply:
According to Ryan, we added the elaboration "T could be instantiated with a different subtype" because people didn't understand exactly why they were getting a (general) assignability error, and it helped people at least in the sense they had a more specific error message they could look up to try and understand the rule. I don't feel it's the same here, the message is already specific about `T` being generic. Do you think the extra information here would help people who don't understand the issue already?